packages/CollapsibleChecklists/src/components/Group/Group.js

```diff
@@ -25,29 +29,59 @@ function useIsIndeterminate(checkboxRef) {
   }, [checkboxRef.current.indeterminate]);
 }
 
-function Group(props) {
-  const { children, isDisabled, onExpand, title } = props;
-  const onChange = React.useContext(CollapsibleChecklistsContext);
-  const [isCollapsed, setIsCollapsed] = React.useState(true);
-  const checkboxRef = React.useRef({});
-  useIsIndeterminate(checkboxRef);
-  let allAreChecked = React.Children.count(props.children) > 0;
-  let noneAreChecked = true;
+function setIsIndeterminateAndGetAllChildItemsAreChecked(
```

Comment:
there are two function here `setIsIndeterminate` And `GetAllChildItemsAreChecked` let's split them to make it clearer what is happening.
```js
function setIsInderterminate(...params) {

}

function areAllChildItemsAreChecked(...params) {
   if(allChildItemsAreChecked) {
     setIsInderterminate(ref, false)
     return;
   }

   setIsInderterminate(ref, true);
}
```